solver/edge.go

```diff
@@ -387,156 +377,194 @@ func (e *edge) markFailed(f *pipeFactory, err error) {
 	})
 }
 
+func (e *edge) processUpdates(updates []pipe.Receiver[*edgeRequest, any]) {
+	depChanged := false
+	for _, upt := range updates {
+		depChanged = e.processUpdate(upt) || depChanged
```

Comment:
This is different behvior that captures last return value.

Reply:
This is likely the cause of the test failure then but I did not realize an or operation in Go could short circuit right to left. I figured it was the same as most languages where this was always evaluated left to right.

Reply:
Ah, I looked over the `||` first time and just saw the variable setting happened on each iteration. This looks correct to me for return value, but it does exec `true = true` in a loop if the first item happens to be return true.

Reply:
Figured out what the original issue was. I misread a condition in creating the slow cache request function. I thought it was `e.preprocessFunc(dep) != nil && e.slowCacheFunc(dep) != nil` but it was an `||` instead.

I believe this logic is correct. I can change it to be something like:

```
if changed := e.processUpdate(upt); changed {
    depChanged = true
}
```

But I thought the extra brevity may be easier to understand. Not a substantial difference between the two though.